test/test.make

```diff
@@ -22,14 +22,19 @@ fmt:
 
 # This ensures that the vendor directory and vendor-bom.csv are in sync
 # at least as far as the listed components go.
+#
+# "go list -m all" returns all modules we depend on, even if we don't
+# actually use any package from it. "go mod vendor" only vendors
+# the packages we really need and we can limit the
+# check to content that we really have in vendor.
 .PHONY: test_vendor_bom
 test: test_vendor_bom
 test_vendor_bom:
 	@ if ! diff -c \
 		<(tail -n +2 vendor-bom.csv | sed -e 's/;.*//') \
-		<((grep '^  name =' Gopkg.lock  | sed -e 's/.*"\(.*\)"/\1/') | LC_ALL=C LANG=C sort); then \
+		<(for path in $$(go list -f '{{.Path}}' -m all); do if [ -d "vendor/$$path" ]; then echo "$$path"; fi; done | LC_ALL=C LANG=C sort); then \
```

Comment:
I started wondering whether we need `$(GO)` (and thus `GOFLAGS=-mod=vendor`) here instead of just go. Sure enough, using `go` results in downloading packages (https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/PR-461/14/pipeline/51).

There's just one snag: with `GOFLAGS=-mod=vendor`, some modules aren't listed anymore (for example, k8s.io/kubectl) even though we have that module in `vendor`. This needs further investigation, don't merge.

Reply:
Raised the question upstream: https://github.com/golang/go/issues/35589

Reply:
@avalluri: please remove this commit and instead cherry-pick this one: https://github.com/pohly/pmem-CSI/commit/43dfe7fabfcd116b947ef6986eda8d5f4ccc3648

I switched to a different approach. Does that make sense?

Reply:
Yes, sound more promising than depending on `go list`. I will cherry-pick.

Reply:
> @avalluri: please remove this commit and instead cherry-pick this one: [pohly@43dfe7f](https://github.com/pohly/pmem-CSI/commit/43dfe7fabfcd116b947ef6986eda8d5f4ccc3648)
> 
Done